Makefile.am

```diff
@@ -294,18 +297,63 @@ libpsmx_fi_la_LDFLAGS = \
 libpsmx_fi_la_LIBADD = $(linkback) $(psm_LIBS)
 libpsmx_fi_la_DEPENDENCIES = $(linkback)
 else !HAVE_PSM_DL
-src_libfabric_la_SOURCES += $(_psm_files)
-src_libfabric_la_CPPFLAGS += $(psm_CPPFLAGS)
-src_libfabric_la_LDFLAGS += $(psm_LDFLAGS)
-src_libfabric_la_LIBADD += $(psm_LIBS)
+noinst_LTLIBRARIES += libpsmx.la
```

Comment:
Yes. The use of convenience libraries allows setting different CPPFLAGS and thus using different psm headers for PSM1 and PSM2.